samples/sid_end_device/include/cli/app_shell_events.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef APP_DUT_H
+#define APP_DUT_H
+
+#include <sidewalk.h>
+
+void dut_sidewalk_event_init_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_deinit_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_start_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_stop_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_get_mtu_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_get_time_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_get_status_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_get_option_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_set_option_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_set_dest_id_handler(void *ctx, void *state_machine);
+void dut_sidewalk_event_set_conn_req_handler(void *ctx, void *state_machine);
+
+#define DUT_EVENT_INIT                                                                             \
+	APP_SIDEWALK_EVENT(DUT_EVENT_INIT, { [STATE_SIDEWALK] = dut_sidewalk_event_init_handler })
```

Comment:
I don't understand , to avoid using enums, you create an event mechanism based on fuctions pointers, but you add another enums (defines). This seems to be more compincated, not simplified.

Reply:
This is a definition of an event. It can be emitted in any state, but different state will handle this event differently.
That is why Event defines the handle for every possible state (if it is needed, if the event must be ignored just ommit the handler for that state)